src/app/dynamic-forms/service/cinchy-dropdown-dataset/cinchy-dropdown-dataset.service.ts

```diff
@@ -111,39 +116,93 @@ export class DropdownDatasetService {
 
     let optionArray: DropdownOption[] = [];
 
-    (await this._cinchyService.executeCsql(dataSetQuery, params).toPromise()).queryResult.toObjectArray().forEach(function (row) {
+    try {
+      (await this._cinchyService.executeCsql(dataSetQuery, params).toPromise()).queryResult.toObjectArray().forEach(function (row) {
 
-      // TODO For now only showing one display column (display-0)
-      const label = row["display-0"] ? `${row["Label"]}, ${row["display-0"]}` : row["Label"];
+        let label = row["Label"];
 
-      optionArray.push(new DropdownOption(row["Id"].toString(), row["Label"]?.toString(), label));
-    });
+        let displayIndex = 0;
 
-    const result = new DropdownDataset(optionArray);
+        do {
+          if (row[`display-${displayIndex}`]) {
+            label += `, ${row[`display-${displayIndex}`]}`;
+          }
+          else {
+            break;
+          }
+        }
+        while (++displayIndex);
```

Comment:
Just a suggestion, I think writing the while loop like this makes it easier to understand when the loop stops running
```suggestion
while (row[`display-${displayIndex}`]) {
    label += `, ${row[`display-${displayIndex}`]}`;
    displayIndex++;
}
```